datafusion/src/execution/context.rs

```diff
@@ -916,35 +898,6 @@ mod tests {
         Ok(())
     }
 
-    #[tokio::test]
-    async fn create_variable_expr() -> Result<()> {
```

Comment:
I think these were added in https://github.com/apache/arrow/pull/8135 . Perhaps @wqc200 has some comment about how / if this feature is used?

Reply:
I don't think we should remove support for scalar variables in the logical plan. This seems unrelated to changing the default SQL dialect. Users can use ScalarVariable without using SQL.